internal/error_test.go

```diff
@@ -441,6 +441,102 @@ func Test_CanceledError(t *testing.T) {
 	require.Equal(t, testErrorDetails3, b3)
 }
 
+// TestCanceledErrorIs tests whether
+// the errors.Is() method operates as
+// expected.
+func TestCanceledErrorIs(t *testing.T) {
+	workflowErr := NewCanceledError()
+	require.True(t, errors.Is(workflowErr, &CanceledError{}), workflowErr)
+}
+
+// TestPanicErrorIs tests whether
+// the errors.Is() method operates as
+// expected.
+func TestPanicErrorIs(t *testing.T) {
+	workflowErr := newPanicError("test value", "stacktrace")
+	require.True(t, errors.Is(workflowErr, &PanicError{}), workflowErr)
+}
+
+// TestApplicationErrorIs tests whether
+// the errors.Is() method operates as
+// expected.
+func TestApplicationErrorIs(t *testing.T) {
+	workflowErr := NewApplicationError("test value", "errType", false, errors.New("error"))
+	require.True(t, errors.Is(workflowErr, &ApplicationError{}), workflowErr)
+}
+
+// TestTimeoutErrorIs tests whether
+// the errors.Is() method operates as
+// expected.
+func TestTimeoutErrorIs(t *testing.T) {
+	workflowErr := NewTimeoutError("timeout", enumspb.TIMEOUT_TYPE_START_TO_CLOSE, nil)
+	require.True(t, errors.Is(workflowErr, &TimeoutError{}), workflowErr)
+}
+
+// TestWorkflowPanicErrorIs tests whether
+// the errors.Is() method operates as
+// expected.
+func TestWorkflowPanicErrorIs(t *testing.T) {
+	workflowErr := newWorkflowPanicError("", "")
+	require.True(t, errors.Is(workflowErr, &workflowPanicError{}), workflowErr)
```

Comment:
You can instead call a little shorter form, which under the hood calls `errors.Is` anyways:
'''
require.ErrorIs(t, workflowErr, &CanceledError{})
'''

Reply:
This is specifically testing that errors.Is(), as the original issue stated. require.ErrorIs() would obfuscate the purpose of the test, no?

Reply:
I would prefer a shorter version given that it does the same thing, but I'll leave this up to you to decide.

Reply:
It appears the user's code originally failed on your more succinct version. Changed the tests!